### ISSUE TITLE
r.drain [Bug] 

### ISSUE THREAD

**Issue by bonushenricus** [OSGeo/grass#3102]
**Describe the bug**
subprocess ended with a non-zero

**To Reproduce**

1. normal DEM 2d raster, EPSG:32632
2. `r.drain input=dem@granello output=drain1 start_coordinates=687115.2159045486,4910484.277135686`
3. 

> Reading input elevation raster map...
> Filling sinks...
> Determining flow directions for ambiguous cases...
> ERROR: Raster map <dem@granello_fill_17622> is not in the current mapset (granello)
> Traceback (most recent call last):
>   File "/usr/local/grass83/scripts/r.drain", line 189, in
> <module>
>     sys.exit(main())
>              ^^^^^^
>   File "/usr/local/grass83/scripts/r.drain", line 149, in
> main
>     grass.run_command(
>   File "/usr/local/grass83/etc/python/grass/script/core.py",
> line 466, in run_command
>     return handle_errors(returncode, result=None, args=args,
> kwargs=kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> ^^^^^^^^^^^^^^^
>   File "/usr/local/grass83/etc/python/grass/script/core.py",
> line 345, in handle_errors
>     raise CalledModuleError(module=module, code=code,
> returncode=returncode)
> grass.exceptions.CalledModuleError: Module run `r.fill.dir
> -f input=dem@granello output=dem@granello_fill_17622
> direction=dem@granello_tmp_17622
> areas=dem@granello_area_17622 format=grass` ended with an
> error.

4. 

> ERROR: dem@granello_tmp_17622: Cannot remove or exclude files not in the current mapset.

**System description (please complete the following information):**

- Operating System: Linux Debian 12
- GRASS GIS version 8.3.0
- python3-wxgtk4.0 4.2.0
- python3 3.11.2

**Comment by bonushenricus** [comment]
I try with the "default" installation of GRASS 8.2.1 in Linux. Same error

**Comment by petrasovaa** [comment]
This was just recently fixed in #3092. A simple workaround is to omit the mapset name in input:
```
r.drain input=dem output=drain1 start_coordinates=687115.2159045486,4910484.277135686
```
